### PR TITLE
Move platform configuration to config files

### PR DESCRIPTION
**Pull request by ktrzcinx**
Move platform configuration to separate file. New platform addition shouldn't be related with rimage recomplilation. Moreover new way of configuration allow easily add default value to any field.

There are a few possible config file formats available to choose:

ini - problem with nested structures
json - lack of comments and poor diff tool output
yaml - advanced language, complicated parsers
toml - supported comments, any problem with nested structures, nice diff tool output

Build test results: https://github.com/thesofproject/sof/pull/3425

### CODE REVIEW

File: src/adsp_config.c [hunk 1]
```diff
@@ -0,0 +1,1158 @@
...
+
+	/* check "version" key */
+	arr = toml_array_in(toml, "version");
+	if (!arr || toml_array_type(arr) != 'i'  || toml_array_nelem(arr) != 2 ||
+	    toml_array_kind(arr) != 'v') {
+		ret = log_err(-EINVAL, "error: toml parse key 'version' failed\n");
+		goto error;
+	}
+
+	for(i = 0; i < ARRAY_SIZE(css_version); ++i){
```
Comment (lgirdwood) [inline src/adsp_config.c:1107]: Missing space after for(

File: src/adsp_config.c [hunk 2]
```diff
@@ -0,0 +1,1158 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```
Comment (lgirdwood) [inline src/adsp_config.c:1]: This file needs a lot more inline comments within each function explaining what is going on - difficult to review otherwise.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
I think the TOML format looks good and is easy to follow.
@jajanusz any comments here on how cmake deal with git submodules ? i.e. SOF has rimage submodule and now rimage has TOML submodule. I guess cmake needs to know how to take care of this ?

File: src/adsp_config.c (hunk 2 above)
Comment (ktrzcinx) [reply src/adsp_config.c:1]: Added much more comments :)

File: src/adsp_config.c (hunk 1 above)
Comment (ktrzcinx) [reply src/adsp_config.c]: Added, btw shouldn't we add checkpach to PR in this repo ?

**Comment by ktrzcinx** [comment]
@lgirdwood Should I redirect git-submodule with toml parser to thesofproject mirror of tomlc99 repo ?

File: src/adsp_config.c (hunk 1 above)
Comment (lgirdwood) [reply src/adsp_config.c]: Sounds like a good idea

**Comment by lgirdwood** [comment]
> @lgirdwood Should I redirect git-submodule with toml parser to thesofproject mirror of tomlc99 repo ?

No we should use the default toml repo. Do we have any patches on top of TOML  ?

**Comment by ktrzcinx** [comment]
>Do we have any patches on top of TOML ?

No, we don't.

**Comment by lgirdwood** [comment]
Ok, use the official toml repo. We can always tag a version if needed.

**Comment by lgirdwood** [comment]
@ktrzcinx do you have a ETA of when we can remove the DNM tag ?

**Comment by ktrzcinx** [comment]
There's no manifest v1.5 support yet, but we can merge it separately, this PR is big enough :)

**Comment by lgirdwood** [comment]
ok, so this wont break any existing cavs1.5 platforms ?

**Comment by ktrzcinx** [comment]
Old way of defining platform is not deleted, so it shouldn't break anything

**Review by lgirdwood — APPROVED** [review]

**Comment by marc-hb** [comment]
Proposal to drop submodules and replace with "west": https://github.com/orgs/thesofproject/teams/sof-developers/discussions/46

**Pull request merged**